core/services/src/main/java/org/exoplatform/services/wcm/portal/listener/CreateLivePortalEventListener.java

```diff
@@ -178,4 +184,18 @@ public UserPortalConfigService getPortalConfigService() {
     }
     return portalConfigService;
   }
+
+  private void initNodeHierarchyCreator() {
+    if (!isNodeHierarchyCreatorInitialized) {
+      LOG.info("init nodeHierarchyCreator ...");
+      try {
+        this.nodeHierarchyCreator.init();
+        isNodeHierarchyCreatorInitialized = true;
+        LOG.info("nodeHierarchyCreator initialized");
+
+      } catch (Exception e) {
+        LOG.error("error when initializing nodeHierarchyCreator", e);
+      }

```

Comment:
Do we need to init the nodeHierarchyCreator ? 
As it is passed as parameter, it should be started and initialized when we arrive to this point ?

Reply:
the CreateLivePortalEventListener starts after the creation of portals at the gatein-portal level with a Startable service that has priority to start before other services so we need to force the start of the nodeHierarchyCreator service to create the "/sites"